### PR TITLE
fix(logs): set logger for connectionDetail page

### PR DESCRIPTION
**Pull request by oceanlvr**
I set logger for the fellow situation

* Message publish success
* Message receive
* Script/Import/Export data status 
* MQTT.js status record

Parts not recorded as logs

* UI operation
* Some operation  just affect the view module data (such as filter out some data by condition)

### CODE REVIEW

**Comment by oceanlvr** [comment]
I set some `TODO/FIXME`  TAGs,  I'm not sure what to do with them. I want to ask for your advice, Thanks!

File: src/views/connections/ConnectionsDetail.vue [hunk 1]
```diff
@@ -282,6 +282,7 @@ import { ipcRenderer } from 'electron'
 import mqtt, { MqttClient, IClientOptions } from 'mqtt'
 import { v4 as uuidv4 } from 'uuid'
 import _ from 'lodash'
+import { getlogger } from '@/utils/logger'
```
Comment (ysfscream) [inline src/views/connections/ConnectionsDetail.vue:285]: Please use camel case to name, I ignored it in my previous review, sorry.

```diff
+ getLogger
- getlogger
```

File: src/views/connections/ConnectionsDetail.vue [hunk 2]
```diff
@@ -510,13 +511,15 @@ export default class ConnectionsDetail extends Vue {
 
   // Connect
   public connect(): boolean | void {
+    const logger = getlogger('connectionDetail', 'debug')
```
Comment (ysfscream) [inline src/views/connections/ConnectionsDetail.vue:514]: I think it’s necessary to refactor here, repeat it many times.

File: src/views/connections/ConnectionsDetail.vue [hunk 3]
```diff
@@ -541,13 +545,14 @@ export default class ConnectionsDetail extends Vue {
           this.$emit('delete')
           this.$message.success(this.$t('common.deleteSuccess') as string)
           this.removeActiveConnection({ id: res.id as string })
+          logger.info(`MQTTX remove connection ${res.id} success`)
```
Comment (ysfscream) [inline src/views/connections/ConnectionsDetail.vue:548]: This `id` has been hidden all the time, and it is impossible to accurately identify which connection it is. Maybe you should add the connection `name` to display.

File: src/views/connections/ConnectionsDetail.vue [hunk 4]
```diff
@@ -775,7 +793,9 @@ export default class ConnectionsDetail extends Vue {
   }
   // Delete topic item
   private handleTopicDelete() {
+    const logger = getlogger('connectionDetail', 'debug')
     this.getMessages()
+    logger.info('Delete the connection topic')
```
Comment (ysfscream) [inline src/views/connections/ConnectionsDetail.vue:798]: We should be able to know which topic was deleted?

File: src/views/connections/ConnectionsDetail.vue (hunk 2 above)
Comment (oceanlvr) [reply src/views/connections/ConnectionsDetail.vue:514]: I think so, I think maybe we can set one `logger` to one `page`.
Init a logger in the `create` event.
what your opinion?

Comment (ysfscream) [reply src/views/connections/ConnectionsDetail.vue:514]: Yes, in the `created` or set a `computed`.

File: src/background.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import { app, protocol, BrowserWindow, ipcMain, shell, Menu } from 'electron'
 import { createProtocol, installVueDevtools } from 'vue-cli-plugin-electron-builder/lib'
 import { updateConnectionMessage } from '@/api/connection'
-import { getlogger, quitAndRenameLogger } from './utils/logger'
+import { getCustomLogger, quitAndRenameLogger } from './utils/logger'
```
Comment (oceanlvr) [inline src/background.ts:6]: `./utils/logger` change to `@/utils/logger` maybe better

**Review by ysfscream — APPROVED** [review]

File: src/components/SubscriptionsList.vue [hunk 6]
```diff
@@ -116,6 +118,10 @@ export default class SubscriptionsList extends Vue {
   @Prop({ required: true }) public connectionId!: string
   @Prop({ required: true }) public record!: ConnectionModel
   @Prop({ type: String, default: '60px' }) public top!: string
+  @Prop({
```
Comment (ysfscream) [inline src/components/SubscriptionsList.vue:121]: Why do you use Prop to initialize the `logger` default value?

**Comment by ysfscream** [comment]
Maybe we can optimize like this, the logger method should be a widely used global function, We can expand into the form of plugins, add some Vue instance methods by attaching them to Vue.prototype.

main.js
```js
Vue.use(Logger)

new Vue({
  //... options
})
```

component.vue
```js
this.$logger.info('debug')
```

Then you can call logger function at will in each component, and we only need to initialize it once, before Vue is constructed.

ref:
https://vuejs.org/v2/guide/plugins.html#Writing-a-Plugin

**Pull request merged**